pandas/core/arrays/base.py

```diff
@@ -1119,3 +1119,148 @@ def _create_arithmetic_method(cls, op):
+                             "size ({size})".format(shape=shape,
+                                                    size=self.size))
+        return type(self)(self, shape=shape)
+
+    def transpose(self, *args, **kwargs):
+        # Note: no validation on args/kwargs
+        shape = self.shape[::-1]
+        return type(self)(self, shape=shape)
+
+    @property
+    def T(self):
+        return self.transpose()
+
+    def ravel(self, order=None):
+        # Note: we ignore `order`, keep the argument for compat with
+        #  numpy signature
+        slen = super().__len__()
+        shape = (slen,)
+        return type(self)(self, shape=shape)
+
+    def swapaxes(self, axis1, axis2):
+        if axis1 > self.ndim or axis2 > self.ndim:
+            raise ValueError("invalid axes=({axis1}, {axis2}) for ndim={ndim}"
+                             .format(axis1, axis2=axis2, ndim=self.ndim))
+        if self.ndim == 1 or axis1 == axis2:
+            return type(self)(self, shape=self.shape)
+        return type(self)(self, shape=self.shape[::-1])
+
+    # --------------------------------------------------------------
+
+    def __getitem__(self, key):
+        if self.ndim == 1:
+            return super().__getitem__(key)
+
+        if not isinstance(key, tuple) or len(key) != 2:
+            raise ValueError("Only limited selection is available for 2D "
+                             "{cls}.  Indexer must be a 2-tuple."
+                             .format(cls=type(self).__name__))
+
+        dummy_dim = 0 if self.shape[0] == 1 else 1
+        oth = 1 - dummy_dim
+
+        key1 = key[dummy_dim]
+        key2 = key[oth]
+        squeezed = self.ravel()
+
+        # Note: we use type(self) explicitly below because
+        #  Arrow bool example otherwise returns numpy
+        if isinstance(key1, int) and key1 == 0:
+            # dummy-dim selection reduces dimension
+            res = squeezed[key2]
+            if np.ndim(res) == 0:
+                return res
+            return type(self)(res)
+        if isinstance(key1, slice):
+            res1d = type(self)(squeezed[[key2]])
+            result = res1d.reshape(1, -1)
+            if dummy_dim == 1:
+                result = result.T
+            return result
+
+        raise NotImplementedError
+
+    def take(self, indices: Sequence[int], allow_fill: bool = False,
+             fill_value: Any = None, axis: int = 0) -> ABCExtensionArray:
+        if self.ndim == 1:
+            return super().take(indices=indices, allow_fill=allow_fill,
+                                fill_value=fill_value)
+
+        def take_item(n):
+            if axis == 0:
+                if n == -1:
+                    seq = [fill_value] * self.shape[1]
+                    return type(self)._from_sequence(seq)
+                else:
+                    return self[n, :]
+            if n == -1:
+                seq = [fill_value] * self.shape[1]
```

Comment:
i think this should be self.shape[0].  either that or above on 1250 it should be self.shape[0]